guide/java/common-usage.md

```diff
@@ -7,134 +7,197 @@ layout: website-normal
+  entity)
+* *Cluster* - a homogeneous collection of entities (all managed by the "cluster" entity)
+* *Fabric* - a multi-location collection of entities, with one per location; often used with a cluster per location
+* *Application* - a top-level entity, which can have one or more child entities.
 
-<!---
-TODO
--->
+The following constructs are often used for Java entities:
 
-- *entity spec* defines an entity, so that one or more such entities can be created; often used by clusters/groups to define how to instantiate new children.
-- *entity factories* are often used by clusters/groups to define how to instantiate new children.
-- *traits* (mixins) providing certain capabilities, such as Resizable and Balanceable
-- *Resizable* entities can re-sized dynamically, to increase/decrease the number of child entities.
-- *Movable* entities can be migrated between *balanceable containers*.
-- *Balanceable containers* can contain *movable* entities, where each contained entity is normally associated with
-    a piece of work within that container.
+* *entity spec* defines an entity to be created; used to define a child entity, or often to 
+  define the type of entity in a cluster.
+* *traits* (mixins) providing certain capabilities, such as *Resizable* and *Startable*.
+* *Resizable* entities can re-sized dynamically, to increase/decrease the number of child entities.
+  For example, scaling up or down a cluster. It could similarly be used to vertically scale a VM,
+  or to resize a disk.
+* *Startable* indicates the effector to be executed on initial deployment (`start()`) and on 
+  tear down (`stop()`).
 
-### Off-the-Shelf Entities
 
-brooklyn includes a selection of entities already available for use in applications,
-including appropriate sensors and effectors, and in some cases include Cluster and Fabric variants.
-(These are also useful as templates for writing new entities.)
- 
-These include:
+### Configuration
 
-- **Web**: Tomcat, JBoss, Jetty (external), Play (external); nginx; GeoScaling
-- **Data**: MySQL, Redis, MongoDB, Infinispan, GemFire (external)
-- **Containers**: Karaf
-- **Messaging**: ActiveMQ, Qpid, Rabbit MQ
-- **PaaS**: Cloud Foundry, Stackato; OpenShift
+Configuration keys are typically defined as static named fields on the Entity interface. These
+define the configuration values that can be passed to the entity during construction. For
+example:
 
+{% highlight java %}
+public static final ConfigKey<String> ROOT_WAR = new ConfigKeys.newStringConfigKey(
+        "wars.root",
+        "WAR file to deploy as the ROOT, as URL (supporting file: and classpath: prefixes)");
+{% endhighlight %}
 
-### Sensors
+One can optional define a `@SetFromFlag("war")`. This defines a short-hand for configuring the
+entity. However, it should be used with caution - the long form defined in the constructor should
+be meaningful and sufficient. Its use may be deprecated in a future release.
```

Comment:
I think it would be better to put it's just not recommended, presumably anything _could_ be deprecated in a future release, is this actually planned?